src/test/java/jenkins/plugins/git/GitToolChooserTest.java

```diff
@@ -0,0 +1,495 @@
+
+        buildAProject(sampleRepo, false);
+
+        List<TopLevelItem> list = jenkins.jenkins.getItems();
+
+        // Assuming no tool is installed and git is present in the machine
+        String gitExe = tool.getGitExe();
+
+        GitToolChooser sizeEstimator = new GitToolChooser(remote, list.get(0), "github", gitExe);
+        assertThat(sizeEstimator.getGitTool(), is("/usr/bin/git"));
+    }
+    /*
+    A test extension implemented to clone the behavior of a plugin extending the capability of providing the size of
+    repo from a remote URL of "Github".
+     */
+    @TestExtension
+    public static class TestExtensionGithub extends GitToolChooser.RepositorySizeAPI {
+
+        @Override
+        public boolean isApplicableTo(String remote, Item context, String credentialsId) {
+            return remote.contains("github");
+        }
+
+        @Override
+        public Long getSizeOfRepository(String remote, Item context, String credentialsId) {
+            // from remote, remove .git and https://github.com
+            return (long) 500;
+        }
+    }
+
+    /*
+    A test extension implemented to clone the behavior of a plugin extending the capability of providing the size of
+    repo from a remote URL of "GitLab".
+     */
+    @TestExtension
+    public static class TestExtensionGitlab extends GitToolChooser.RepositorySizeAPI {
+
+        @Override
+        public boolean isApplicableTo(String remote, Item context, String credentialsId) {
+            return remote.contains("gitlab");
+        }
+
+        @Override
+        public Long getSizeOfRepository(String remote, Item context, String credentialsId) {
+            // from remote, remove .git and https://github.com
+            return (long) 10000;
+        }
+    }
+
+    /*
+    A test extension implemented to clone the behavior of a plugin extending the capability of providing the size of
+    repo from a remote URL of "BitBucket".
+     */
+    @TestExtension
+    public static class TestExtensionBit extends GitToolChooser.RepositorySizeAPI {
+
+        @Override
+        public boolean isApplicableTo(String remote, Item context, String credentialsId) {
+            return remote.contains("bit");
+        }
+
+        @Override
+        public Long getSizeOfRepository(String remote, Item context, String credentialsId) throws IOException {
+            throw new IOException();
+        }
+    }
+
+
+    private void buildAProject(GitSampleRepoRule sampleRepo, boolean noCredentials) throws Exception {
+        WorkflowJob p = jenkins.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n"
+                        + "  checkout(\n"
+                        + "    [$class: 'GitSCM', \n"
+                        + "      userRemoteConfigs: [[credentialsId: 'github', url: $/" + sampleRepo + "/$]]]\n"
```

Comment:
Use the credentials that were conditionally defined earlier.
```suggestion
                        + "      userRemoteConfigs: [[" + creds + "url: $/" + sampleRepo + "/$]]]\n"
```